src/shared/services/shortcut-action.service.ts

```diff
@@ -0,0 +1,232 @@
+   * @param {number} direction - Movement direction. -1 for UP, +1 for DOWN
+   * @returns {number} - Returns the new index of the moved element
+   */
+  private moveElement(index: number, direction: number, path: Array<any>): number {
+    let values =  this.jsonStoreService.getIn(path);
+    let newIndex = ((index + direction) < values.size &&
+                    (index + direction) >= 0) ? index + direction : values.size - Math.abs((index + direction));
+    let temp = values.get(index);
+    values = values
+      .set(index, values.get(newIndex))
+      .set(newIndex, temp);
+    this.jsonStoreService.setIn(path, values);
+    return newIndex;
+  }
+
+  deleteAction(path: Array<any>) {
+    if (window.confirm('Are you sure you want to delete this item?')) {
+      this.deleteElement(this.pathUtilService.getNearestOrRootArrayParentInPath(path, false),
+        this.pathUtilService.getElementIndexInForwardOrReversePath(path, false));
+    }
+  }
+
+  /**
+   * @param {Array<any>} path - Path of the element's array parent
+   * @param {number} index - Index of the element that is deleted from array parent path
+   */
+  private deleteElement(path: Array<any>, index: number) {
+    let values = this.jsonStoreService.getIn(path);
+    this.jsonStoreService.setIn(path, values.delete(index));
+    this.tabIndexService.deleteElemTabIndex(path.join('.'));
+  }
+
+  navigateUpAction(path: Array<any>): void {
+    this.navigateUpDown(path, -1);
+  }
+
+  navigateDownAction(path: Array<any>): void {
+    this.navigateUpDown(path, 1);
+  }
+
+  /**
+   * @param {Array<any>} path - Path of the element that is focused
+   * @param {number} direction - Navigation direction. -1 for UP, +1 for DOWN
+   */
+  private navigateUpDown(path: Array<any>, direction: number) {
+    let values = this.jsonStoreService.getIn(this.pathUtilService.getNearestOrRootArrayParentInPath(path, false));
+    if (List.isList(values)) {
+      let elemIndexInPath = this.pathUtilService.getElementIndexInForwardOrReversePath(path, true);
+      path[path.length - 2] = ((elemIndexInPath + direction) < values.size &&
+                              (elemIndexInPath + direction) >= 0) ? elemIndexInPath + direction :
+                              values.size - Math.abs((elemIndexInPath + direction));
+      this.domUtilService.focusAndSelectFirstInputChildById(path.join('.'));
+    }
+  }
+
+  navigateLeftAction(path: Array<any>): void {
+    this.navigateRightLeft(path, -1);
+  }
+
+  navigateRightAction(path: Array<any>): void {
+    this.navigateRightLeft(path, 1);
+  }
+
+  /**
+   * @param {Array<any>} path - Path of the element that is focused
+   * @param {number} direction - Navigation direction. -1 for LEFT, +1 for RIGHT
+   */
+  private navigateRightLeft(path: Array<any>, direction: number) {
+    let tabIndex = this.tabIndexService.getElemTabIndex(path.join('.'));
+    if (tabIndex >= 0) {
+      this.domUtilService.focusRightOrLeftCell(tabIndex, direction);
+    }
```

Comment:
first see the comment for `domUtilService.focusRightOrLeftCell`.

After that you can reduce this, to

`this.domUtilService.focusRightOrLeftParentCellById(path.join('.'), direction);`